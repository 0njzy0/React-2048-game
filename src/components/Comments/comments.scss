@import '../../assets/styles/vars';

.main {
  text-align: center;
}

.comments {
  background: $gray;
  border-radius: 5px;
  margin: 3.85vw auto 0;
  padding: 1.54vw;
  width: 60vw;

  .comments-title {
    border-bottom: 1px solid $border-color;
    font-size: 1.25rem;
    font-weight: 500;
    padding-bottom: 0.77vw;
    text-align: left;

    span {
      background: $gray;
      color: $light-black;
      text-decoration: line-through;
    }
  }
}

