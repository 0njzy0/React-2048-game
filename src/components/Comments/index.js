import React, { Component } from 'react';
import Gitalk from 'gitalk';
import styles from './comments.scss';
import i18n from '../../utils/i18n';

const gitalk = new Gitalk({
  clientID: '3bff581d9182d38a4598',
  clientSecret: 'f49cd76c182957bbcd2593b6d7bce5c6ae69b384',
  repo: 'React-2048-game',
  owner: 'devrsi0n',
  admin: ['devrsi0n'],
  // facebook-like distraction free mode
  distractionFreeMode: false,
  labels: ['gitment', 'https://devrsi0n.github.io/React-2048-game/']
});

export default class Comments extends Component {
  componentDidMount() {
    if (process.env.NODE_ENV !== 'test') {
      gitalk.render('gitalk-container');
    }
  }

  render() {
    return (
      <div className={styles.main}>
        <div className={styles.comments}>
          {/* eslint-disable react/no-danger */}
          <p
            className={styles.commentsTitle}
            dangerouslySetInnerHTML={{ __html: i18n.commentTitle }}
          />
          <div id="gitalk-container" className={styles.container} />
        </div>
      </div>
    );
  }
}
