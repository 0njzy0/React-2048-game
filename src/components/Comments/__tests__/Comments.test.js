import React from 'react';
import renderer from 'react-test-renderer';
import Comments from '..';

// Ignore gitment error message
console.error = () => {};

describe('<Comments />', () => {
  it('component render', () => {
    const comments = renderer.create(<Comments />).toJSON();
    expect(comments).toMatchSnapshot();
  });
});
