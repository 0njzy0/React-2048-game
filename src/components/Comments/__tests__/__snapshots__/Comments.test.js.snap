// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> component render 1`] = `
<div
  className="main"
>
  <div
    className="comments"
  >
    <p
      className="commentsTitle"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Welcome to leave comments",
        }
      }
    />
    <div
      id="gitmentHeader"
    />
    <div
      id="gitmentComments"
    />
    <div
      id="gitmentEditor"
    />
  </div>
  <div
    className="footer"
  >
    <a
      className="icon"
      href="https://github.com/devrsi0n/React-2048-game"
    >
      <img
        alt="github repo"
        className="github"
        src="github.png"
      />
    </a>
    Build with
    <span
      className="heart"
    >
      â™¥
    </span>
    by
    <a
      className="link"
      href="https://github.com/devrsi0n"
    >
      devrsi0n
    </a>
  </div>
</div>
`;
